cmd/testpeers.go

```diff
@@ -204,14 +204,14 @@ func pingPeerOnce(ctx context.Context, tcpNode host.Host, peer p2p.Peer) (ping.R
 
 func pingPeerContinuously(ctx context.Context, tcpNode host.Host, peer p2p.Peer, resCh chan ping.Result) {
 	for {
+		r, err := pingPeerOnce(ctx, tcpNode, peer)
+		if err != nil {
+			return
+		}
 		select {
 		case <-ctx.Done():
 			return
 		default:
-			r, err := pingPeerOnce(ctx, tcpNode, peer)
-			if err != nil {
-				return
-			}
 			resCh <- r
```

Comment:
Now this write does not respect the context and can block the routine.

Reply:
Why it wouldn't respect the context? I do:
```
		select {
		case <-ctx.Done():
			return
		default:
			resCh <- r
			awaitTime := rand.Intn(100) //nolint:gosec // weak generator is not an issue here
			time.Sleep(time.Duration(awaitTime) * time.Millisecond)
		}
```
I still have the context check

Reply:
If `resCh <- r` is blocked then you are stuck, because this code executes not under `select`, but in the code block of the `default` case. It is not a `case` to be simple, therefore `select` does not work for this write.

Reply:
https://goplay.tools/snippet/wS9Sbd8kMu9

Reply:
Looking at it now... It seems to have been an issue even before this change, right?

Reply:
Right